LFPy/test/test_network.py::testNetwork::test_Network_06 FAILED

**Describe the bug**
LFPy/test/test_network.py::testNetwork::test_Network_06 fails on certain hardware/python/h5py versions:

```
        # check that saved connections are indeed correct
        f = h5py.File(os.path.join(network.OUTPUTPATH,
                                   'synapse_connections.h5'), 'r')
        conn_data = f['test:test'][()]
    
        assert np.all(conn_data['weight'] ==
                      connectionParameters['weightargs']['value'])
        assert np.all(conn_data['delay'] ==
                      connectionParameters['delayargs']['value'])
        assert np.all(conn_data['sec.x'] == 0.5)
        for cell in population.cells:
            inds = conn_data['gid'] == cell.gid
            assert np.all(conn_data['gid_pre'][inds] != cell.gid)
            for secname in conn_data['sec'][inds]:
                assert secname.decode() == \
                    'ball_and_stick_template[{}].soma[0]'.format(cell.gid)
            assert np.all(conn_data['x'][inds] == cell.x.mean(axis=-1)[0])
            assert np.all(conn_data['y'][inds] == cell.y.mean(axis=-1)[0])
            assert np.all(conn_data['z'][inds] == cell.z.mean(axis=-1)[0])
    
        # check static connection parameters
        synparams = f['synparams']['test:test']
>       assert synparams['mechanism'][()].decode() == \
            connectionParameters['syntype'].__str__().strip('()')
E       AttributeError: 'str' object has no attribute 'decode'

LFPy/test/test_network.py:717: AttributeError
```
With h5py v2.10.0 `type(synparams['mechanism'][()])` is `str` not `bytes` (thankfully).

--- a/LFPy/test/test_network.py
+++ b/LFPy/test/test_network.py
@@ -714,8 +714,12 @@ class testNetwork(unittest.TestCase):
 
         # check static connection parameters
         synparams = f['synparams']['test:test']
-        assert synparams['mechanism'][()].decode() == \
-            connectionParameters['syntype'].__str__().strip('()')
+        if h5py.__version__ < '3':
+            assert synparams['mechanism'][()] == \
+                connectionParameters['syntype'].__str__().strip('()')
+        else:
+            assert synparams['mechanism'][()].decode() == \
+                connectionParameters['syntype'].__str__().strip('()')
         for key, value in connectionParameters['synparams'].items():
             assert synparams[key][()] == value
 

